js/index.js

```diff
@@ -0,0 +1,217 @@
+}
+
+function storeLibrary() {
+  library.forEach((book, idx) => {
+    localStorage.setItem(Book.dbBookKey(idx), book.json);
+  });
+}
+
+function loadLibrary(addBookFn) {
+  library.length = 0;
+  let idx = 0;
+  let key = Book.dbBookKey(idx);
+  while (localStorage.getItem(key)) {
+    addBookFn(JSON.parse(localStorage.getItem(key)));
+    key = Book.dbBookKey(idx += 1);
+  }
+}
+
+function render(page = currentPage) {
+  const libraryContainer = document.querySelector('#library');
+  const libContainer = document.createElement('div');
+  libContainer.id = 'library';
+  library.slice(page * PAGE_SIZE, (page + 1) * PAGE_SIZE).forEach((book, index) => {
+    const bookElement = book.render(index);
+    libContainer.appendChild(bookElement);
+  });
+  document.querySelector('.main').replaceChild(libContainer, libraryContainer);
+  libContainer.addEventListener('click', (event) => {
+    if (event.target.classList.contains('delete-book')) {
+      const index = event.target.parentElement.id;
+      library.splice(index, 1);
+      if (currentPage > lastPage()) {
+        currentPage = lastPage();
+      }
+      render();
+      storeLibrary();
+      localStorage.removeItem(Book.dbBookKey(library.length));
+    }
+    if (event.target.classList.contains('read-book')) {
+      const index = event.target.parentElement.id;
+      library[index].readStatus = !library[index].readStatus;
+      render();
+      storeLibrary();
+    }
+  });
+}
+
+function addBookToLibrary(book) {
+  library.push(new Book(book));
+}
+
+function cleanFields() {
+  document.querySelectorAll('input[type=text]').forEach((input) => {
+    input.value = '';
+  });
+}
+
+function bringUpForm() {
+  cleanFields();
+  form.classList.add('active');
+}
+
+function processBook(event) {
+  event.preventDefault();
+  const book = {};
+  const fd = new FormData(form);
+  Array.from(fd).forEach(([key, value]) => {
+    book[key] = value;
+  });
+  addBookToLibrary(book);
+  currentPage = lastPage();
+  render();
+  storeLibrary();
+  form.classList.remove('active');
+}
+
+form.addEventListener('submit', processBook);
+newBookBtn.addEventListener('click', bringUpForm);
+
+prevPageBtn.addEventListener('click', () => { if (currentPage > 0) { render(currentPage -= 1); } });
+nextPageBtn.addEventListener('click', () => { if (currentPage < lastPage()) { render(currentPage += 1); } });
+
+firstPageBtn.addEventListener('click', () => { render(currentPage = 0); });
+lastPageBtn.addEventListener('click', () => { render(currentPage = lastPage()); });
+
+if (storageAvailable('localStorage') && localStorage.getItem(Book.dbBookKey(0))) {
+  loadLibrary(addBookToLibrary);
+  currentPage = 0;
+  render();
+} else {
+  fetch(`${BOOKS_URL}/books.json`)
+    .then(response => response.json())
+    .then((data) => {
+      data.forEach((book) => {
+        book.imageLink = `${BOOKS_URL}/static/${book.imageLink}`;
+        addBookToLibrary(book);
+      });
+      render();
+      storeLibrary();
+    });
```

Comment:
This is what is causing the issue. You shouldn't need to pull the books from any endpoint. When you do it will always return even after deleting since you can't mutate what the endpoint returns.

Remove this section of the code and leave the books to be created from the new book form only.